src/test/groovy/graphql/schema/SchemaTransformerTest.groovy

```diff
@@ -160,4 +226,119 @@ type SubChildChanged {
 
 
     }
+
+    def "traverses operation types"() {
+        given:
+        GraphQLSchema schema = TestUtil.schema("""
+        type Query {
+            hello: Foo 
+        }
+        
+        type Subscription {
+            fooHappened: FooEvent
+        }
+        
+        type Mutation {
+            updateFoo(foo: FooUpdate): Boolean
+        }
+        
+        type FooEvent {
+            foo: Foo
+            timestamp: Int
+        }
+        
+        type Foo {
+           bar: Bar
+        } 
+        
+        input FooUpdate {
+            newBarBaz: String
+        }
+        
+        type Bar {
+            baz: String
+        }
+        
+        type Baz {
+            bing: String
+        }
+        
+        """)
+        SchemaTransformer schemaTransformer = new SchemaTransformer()
+
+        when:
+        final Set<String> visitedTypeNames = []
+        schemaTransformer.transform(schema, SchemaTransformer.TraversalType.OPERATION_TYPES_ONLY, new GraphQLTypeVisitorStub() {
+            @Override
+            TraversalControl visitGraphQLObjectType(GraphQLObjectType node, TraverserContext<GraphQLSchemaElement> context) {
+                visitedTypeNames << node.name
+
+                TraversalControl.CONTINUE
+            }
+
+            @Override
+            TraversalControl visitGraphQLInputObjectType(GraphQLInputObjectType node, TraverserContext<GraphQLSchemaElement> context) {
+                visitedTypeNames << node.name
+
+                TraversalControl.CONTINUE
+            }
+        })
+
+        then:
+        visitedTypeNames.contains('Foo')
+        visitedTypeNames.contains('FooUpdate')
+        visitedTypeNames.contains('FooEvent')
+        visitedTypeNames.contains('Bar')
+        !visitedTypeNames.contains('Baz')
+    }
+
+
+    def "transformed schema can be executed"() {
+
+        given:
+        def queryObject = newObject()
```

Comment:
We tend to use SDL in tests and RuntimeWiring.

TestUtil has a way to do it cleanly.

The use of runtime data fetcher wiring should be avoided even in tests.  We are trying to deprecate this capability

Reply:
def sdl = ' type Query  { foo : String }`
def schema = TestUtil.schema(sdl, [ "Query" : [ "foo" : dataFetcherHere ]]):

Reply:
@andimarek wanted to keep a test that uses the query API rather than the SDL, but i've added another test which uses the SDL. In the process we uncovered a bug where field renaming doesn't update the data fetcher, which is now addressed.